app/js/components/ConceptDisplay.jsx

```diff
@@ -12,27 +12,19 @@ const {
 } = formValidations;
 
 class ConceptDisplay extends PureComponent {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      concept: {},
-    };
-  }
-
   UNSAFE_componentWillMount() {
-    const { conceptUUID, dispatch, concept } = this.props;
+    const {
+      conceptUUID, dispatch, concept,
+    } = this.props;
     if (R.isEmpty(concept)) {
       dispatch(fetchConcept(conceptUUID));
     }
   }
 
   UNSAFE_componentWillReceiveProps(nextProps) {
-    const { concept } = this.state;
-    if (nextProps.concept !== concept) {
-      this.setState({
-        concept,
-      });
+    const { onLoaded } = this.props;
+    if (!R.isEmpty(nextProps.concept)) {
+      onLoaded();
```

Comment:
The `ConceptDisplay` reports its loading status back up to the `Cell` which reports its loading status back up to the table. The table resolves a promise when all the cells are loaded.